Lib/inspect.py

```diff
@@ -2203,7 +2203,7 @@ def _signature_strip_non_python_syntax(signature):
         add(string)
         if (string == ','):
             add(' ')
-    clean_signature = ''.join(text).strip()
```

Comment:
This change is because for some reason the inspect module is relying on the fact that if lines yielded by the generator do not end in `\n` then they are concatenated together, which is wrong because the contract says "should yield one line at a time".